erpnext/payroll/doctype/payroll_entry/payroll_entry.py

```diff
@@ -60,6 +60,8 @@ def validate_employee_details(self):
 	def on_cancel(self):
 		frappe.delete_doc("Salary Slip", frappe.db.sql_list("""select name from `tabSalary Slip`
 			where payroll_entry=%s """, (self.name)))
+		self.salary_slips_created = False
+		self.salary_slips_submitted = False
```

Comment:
```suggestion
		self.db_set("salary_slips_created", 0)
		self.db_set("salary_slips_submitted", 0)
```

https://github.com/frappe/erpnext/blob/8f0d10a1b791fb6c9471fe8aec785e5bd6ac5321/erpnext/payroll/doctype/payroll_entry/payroll_entry.py#L699

Use `db_set` while doing operations in `on_cancel` or `on_submit` since it's not explicitly committed to db.

Reply:
Good to know, thanks!